docs/Compose.md

```diff
@@ -0,0 +1,181 @@
+| Ingestor  | Takes completed GUAC objects and performs GraphQL mutation queries to add to GUAC |
+| Assembler | Takes parsed document objects and assembles GUAC objects                          |
+| Parser    | Takes documents (ex: SBOMs) and parses the objects inside them                    |
+| Collector | Reads or watches locations for new documents and collects them when found         |
+
+There are different collectors for the different types of locations that GUAC
+can watch: files, storage buckets, git repositories, etc. There are different
+types of Parser/Assemblers for the different document types that GUAC can
+understand: SPDX, CycloneDX, etc. GUAC uses [Nats](https://nats.io/) for
+communication between components.
+
+### Certifiers
+
+Another class GUAC components are certifiers. Certifiers run outside the server,
+but are not part of the ingestion pipeline. They watch the server for new nodes
+in the server, and then try to add additional information attached to those
+nodes.
+
+For example, the [OSV](https://ossf.github.io/osv-schema/) certifier will watch
+GUAC for new packages, then try to discover OSV vulnerabilities for those
+packages. If any vulnerabilities are found, the certifier will attach a
+"CertifyVuln" node to the package that signifies that the package is connected
+to the OSV vulnerability.
+
+## Prerequisites
+
+- Docker
+- Docker Compose
+- Git
+- Go (optional)
+- Make (optional)
+
+## Clone GUAC
+
+If you haven't already, clone GUAC to a local directory:
+
+```bash
+git clone https://github.com/guacsec/guac.git
+```
+
+Optional: Also clone GUAC data, this is used as test data.
+
+```bash
+git clone https://github.com/guacsec/guac-data.git
+```
+
+The rest of the demo will assume you are in the GUAC directory
+
+```bash
+cd guac
+```
+
+## Build the containers
+
+The `Makefile` contains the `docker build` commands:
+
+```bash
+make container
+```
+
+> Note: you may also run the `docker build` command directly without `make`.
+
+## Start GUAC
+
+In another terminal, start up GUAC.
+
+```bash
+docker-compose up
+```
+
+The full GUAC deployment is now running. The GraphQL server is listening on port
+`8080`. You may visit [http://localhost:8080](http://localhost:8080) to see the
+GraphQL playground. GraphQL queries are served at the `/query` endpoint.
+
+Nats is listening on port `4222`, this is where any collectors that you run will
+need to connect to push any files they find. The GUAC `collector` command
+defaults to `nats://127.0.0.1:4222` for the Nats address, so this will work
+automatically.
+
+## Start Ingesting Data
+
+Now you can run any GUAC collector and push data to the running ingestion
+pipeline. For example we can ingest the sample `guac-data` data. However, you
+may ingest what you wish to here instead.
+
+```bash
+pushd ../guac-data/docs
+docker run --rm -v $PWD:/data --network guac_default local-organic-guac:latest /opt/guac/collector files /data --natsaddr nats:4222
```

Comment:
let's add setup instructions for using `guac-data` repo, or point to an initial repo set up that they should ensure be done before hand. Also would be helpful here to explain parts of the docker command (like the mounting of volumes, etc..

Reply:
👍